lib/galaxy/tools/deps/resolvers/galaxy_packages.py

```diff
@@ -13,8 +13,40 @@
 log = logging.getLogger( __name__ )
 
 
+class GalaxyPackageDependency(Dependency):
+    dict_collection_visible_keys = Dependency.dict_collection_visible_keys + ['script', 'path', 'version', 'name']
+    dependency_type = 'galaxy_package'
+
+    def __init__( self, script, path, version, name, exact=True ):
+        self.script = script
+        self.path = path
+        self.version = version
+        self.name = name
+        self._exact = exact
+
+    @property
+    def exact(self):
+        return self._exact
+
+    def shell_commands( self, requirement ):
+        base_path = self.path
+        if self.script is None and base_path is None:
+            log.warning( "Failed to resolve dependency on '%s', ignoring", requirement.name )
+            commands = None
+        elif requirement.type == 'package' and self.script is None:
+            commands = 'PACKAGE_BASE=%s; export PACKAGE_BASE; PATH="%s/bin:$PATH"; export PATH' % ( base_path, base_path )
+        else:
+            commands = 'PACKAGE_BASE=%s; export PACKAGE_BASE; . %s' % ( base_path, self.script )
+        return commands
+
+
+class ToolShedDependency(GalaxyPackageDependency):
+    dependency_type = 'tool_shed_package'
+
+
 class BaseGalaxyPackageDependencyResolver(DependencyResolver, UsesToolDependencyDirMixin):
     dict_collection_visible_keys = DependencyResolver.dict_collection_visible_keys + ['base_path', 'versionless']
+    DependencyType = GalaxyPackageDependency
```

Comment:
My inclination would be to call this variable ``dependency_type`` - I'm not positive I am right though. Ping @nsoranzo